sgx_libc/src/linux/x86_64/ocall.rs

```diff
@@ -2912,29 +2335,28 @@ pub unsafe fn pipe(fds: *mut c_int) -> c_int {
     result
 }
 
-pub unsafe fn pipe2(fds: *mut c_int, flags: c_int) -> c_int {
+pub unsafe fn pipe2(fds: &mut [c_int; 2], flags: c_int) -> OCallResult<()> {
```

Comment:
`pub unsafe fn pipe2(fds: *mut [c_int; 2], flags: c_ing) -> OCallResult<()>`

maintain a unified style of function definition with pipe.